openhcl/underhill_attestation/src/lib.rs

```diff
@@ -51,16 +53,32 @@ use zerocopy::FromZeroes;
 /// An attestation error.
 #[derive(Debug, Error)]
 #[error(transparent)]
-pub struct Error(ErrorInner);
+pub struct Error(AttestationErrorInner);
 
-impl<T: Into<ErrorInner>> From<T> for Error {
+impl Error {
+    fn log_error_id<T: Any + Debug>(e: &T) {
+        let value_any = e as &dyn Any;
+        let type_name = std::any::type_name::<T>();
+
+        match value_any.downcast_ref::<String>() {
+            Some(as_string) => {
+                tracing::error!(CVM_ALLOWED, "Unexpected string {:?}", as_string);
+            }
+            None => {
+                tracing::error!(CVM_ALLOWED, "AttestationErrorId::{type_name}::{e:?}");
```

Comment:
High-level comments regarding this approach: 
We're mapping the one-level nested error enum from selective module to `AttestationErrorId::` where some of the information are not covered/unsued (take `ReadSecurityProfile(#[source] vmgs::ReadFromVmgsError)` for example, `ReadSecurityProfile` and the nested error within ` vmgs::ReadFromVmgsError` are missing). I'm wonder if we can have a more generic approach as part of the error mapping that we already do such that the multiple-level nested errors can be handled properly.

nit: The error string itself will be duplicated in the logs (this tracing and the Rust default path)

Reply:
When debugging the more information, the better.  it should include the nested errors as well, here is a sample output:
{"message":"AttestationErrorId::underhill_attestation::secure_key_release::RequestVmgsEncryptionKeysError::SendIgvmAttestKeyReleaseRequest(IgvmAgentGenericError)"}